pallets/xtransfer/src/bridge_transfer/mod.rs

```diff
@@ -169,14 +202,24 @@ pub mod pallet {
 
 			let asset_amount = T::BalanceConverter::to_asset_balance(amount, asset_id)
 				.map_err(|_| Error::<T>::BalanceConversionFailed)?;
-			// burn asset from sender
-			<pallet_assets::pallet::Pallet<T> as FungibleMutate<T::AccountId>>::burn_from(
-				asset_id,
-				&source,
-				asset_amount,
-			)
-			.map_err(|_| Error::<T>::FailedToTransactAsset)?;
 
+			if asset_reserve_location == dest_reserve_location {
+				// burn if transfer back to its reserve location
+				pallet_assets::pallet::Pallet::<T>::burn_from(asset_id, &sender, asset_amount)
+					.map_err(|_| Error::<T>::FailedToTransactAsset)?;
+			} else {
+				// transfer asset from sender to reserve account
+				<pallet_assets::pallet::Pallet<T> as FungibleTransfer<T::AccountId>>::transfer(
```

Comment:
```suggestion
				pallet_assets::pallet::Pallet<T>::transfer(
```

Reply:
`pallet_assets::pallet::Pallet<T>::transfer` has different arguments here, and if use it, we have to handle the `AccountId` and `Lookup::Source` conversion.